In [2]:
import sqlite3
import contextlib
import pandas as pd

In [4]:
sql_connect = sqlite3.connect('database.sqlite')
cursor = sql_connect.cursor()

In [39]:
query_r = "SELECT * FROM sqlite_sequence" # read query
query_i = "INSERT INTO country VALUES ('7819', 'mooi land')" # insert query
query_d = "DELETE FROM country WHERE name='mooi land'" # delete query

In [29]:
sql_connect = sqlite3.connect('database.sqlite')
cursor = sql_connect.cursor()
df = pd.read_sql_query(query_r,sql_connect)
df

id         name
0       1      Belgium
1    1729      England
2    4769       France
3    7809      Germany
4   10257        Italy
5   13274  Netherlands
6   15722       Poland
7   17642     Portugal
8   19694     Scotland
9   21518        Spain
10  24558  Switzerland

In [40]:
import contextlib
@contextlib.contextmanager
def db_read_only(file):
    sql_connect = sqlite3.connect('file:'+file+'?mode=ro',uri=True)
    cursor = sql_connect.cursor()
    yield sql_connect
    sql_connect.commit()
    sql_connect.close() 

In [41]:
file = 'database.sqlite'
with db_read_only(file) as ro:
    data = pd.read_sql_query(query_r, ro)

In [42]:
data['name']

0                 Team
1              Country
2               League
3                Match
4               Player
5    Player_Attributes
6      Team_Attributes
Name: name, dtype: object

Gegevens van onze club:

- Team naam: Atlético Madrid
- Team ID: 9906
- Competitie ID: 21518
- Competitie naam: Spain LIGA BBVA
- Gegevens van gespeelde wedstrijden

In [43]:
df_list = []
with db_read_only(file) as ro:
    for table in data['name']:
        query_r = 'SELECT * FROM ' + str(table)
        data = pd.read_sql_query(query_r, ro)
        df_list.append(data)
df_list[0]

id  team_api_id  team_fifa_api_id      team_long_name team_short_name
0        1         9987             673.0            KRC Genk             GEN
1        2         9993             675.0        Beerschot AC             BAC
2        3        10000           15005.0    SV Zulte-Waregem             ZUL
3        4         9994            2007.0    Sporting Lokeren             LOK
4        5         9984            1750.0   KSV Cercle Brugge             CEB
..     ...          ...               ...                 ...             ...
294  49479        10190             898.0       FC St. Gallen             GAL
295  49837        10191            1715.0             FC Thun             THU
296  50201         9777             324.0         Servette FC             SER
297  50204         7730            1862.0  FC Lausanne-Sports             LAU
298  51606         7896               NaN              Lugano             LUG

[299 rows x 5 columns]

In [54]:
team = df_list[0]
country = df_list[1]
league = df_list[2]
match = df_list[3]
player = df_list[4]
player_attributes = df_list[5]
team_attributes = df_list[6]